exist-core/src/test/xquery/xmldb/collection-create-tests.xql

```diff
@@ -0,0 +1,58 @@
+(:
+ : eXist-db Open Source Native XML Database
+ : Copyright (C) 2001 The eXist-db Authors
+ :
+ : info@exist-db.org
+ : http://www.exist-db.org
+ :
+ : This library is free software; you can redistribute it and/or
+ : modify it under the terms of the GNU Lesser General Public
+ : License as published by the Free Software Foundation; either
+ : version 2.1 of the License, or (at your option) any later version.
+ :
+ : This library is distributed in the hope that it will be useful,
+ : but WITHOUT ANY WARRANTY; without even the implied warranty of
+ : MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ : Lesser General Public License for more details.
+ :
+ : You should have received a copy of the GNU Lesser General Public
+ : License along with this library; if not, write to the Free Software
+ : Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ :)
+xquery version "3.0";
+
+module namespace t="http://exist-db.org/testsuite/collection-create";
+
+import module namespace test="http://exist-db.org/xquery/xqsuite" at "resource:org/exist/xquery/lib/xqsuite/xqsuite.xql";
+
+declare variable $t:parent-collection-name := "/parent-collection";
+declare variable $t:parent-collection := "/db" || $t:parent-collection-name;
+declare variable $t:path-collection := $t:parent-collection-name || "/path/to/new-collection";
+declare variable $t:path-collection-from-root := "/db/path/to/new-collection-from-root";
+
+declare
+    %test:setUp
+function t:setup() {
+    xmldb:create-collection("/db", $t:parent-collection-name)
+};
+
+declare
+    %test:tearDown
+function t:cleanup() {
+    xmldb:remove($t:parent-collection),
+    xmldb:remove($t:path-collection-from-root)
+};
+
+declare
+    %test:assertEquals("/db/parent-collection/path/to/new-collection")
+function t:fnCreateNewRecursiveCollection() {
+    let $collection := xmldb:create-collection($t:path-collection)
```

Comment:
Do I understand correctly that you are currently testing that:
```
xmldb:create-collection("/parent-collection/path/to/new-collection")
```
creates the collection `/db/parent-collection/path/to/new-collection`?

If so, that should not work! The `xmldb:create-collection#1` function should always take an absolute path, i.e. starting with `/db`. This test should be modified to check for an error result, and the implementation updated accordingly.

Reply:
looks good to me